src/res.rs

```diff
@@ -263,36 +274,63 @@ impl Resources {
     pub fn fetch_mut<T>(&self, id: usize) -> FetchMut<T>
         where T: Resource
     {
-        let c = self.fetch_internal(TypeId::of::<T>(), id);
+        self.try_fetch_mut(id).expect(RESOURCE_NOT_FOUND)
+    }
 
-        FetchMut {
-            inner: c.borrow_mut(),
-            phantom: PhantomData,
-        }
+    /// Like `fetch_mut`, but returns an `Option` instead of panicking in the case of the resource
+    /// being accessed mutably.
+    pub fn try_fetch_mut<T>(&self, id: usize) -> Option<FetchMut<T>>
+        where T: Resource
+    {
+        self.try_fetch_internal(TypeId::of::<T>(), id)
+            .map(|r| {
+            FetchMut {
+                inner: r.borrow_mut(),
+                phantom: PhantomData,
+            }
+        })
     }
 
     /// Fetches the resource with the specified type id.
     ///
     /// Please see `fetch` for details.
+    ///
+    /// # Panics
+    ///
+    /// Panics if no resource with the id exists.
     pub fn fetch_id(&self, id: TypeId, comp_id: usize) -> FetchId {
-        let c = self.fetch_internal(id, comp_id);
+        let c = self.try_fetch_internal(id, comp_id).expect(RESOURCE_NOT_FOUND);
 
         FetchId { inner: c.borrow() }
     }
 
+    /// Like `fetch_id`, but returns an `Option` rather than panicking.
+    pub fn try_fetch_id(&self, id: TypeId, comp_id: usize) -> Option<FetchId> {
+        self.try_fetch_internal(id, comp_id).map(|r| FetchId { inner: r.borrow() })
+    }
+
     /// Fetches the resource with the specified type id mutably.
     ///
     /// Please see `fetch` for details.
+    ///
+    /// # Panics
+    ///
+    /// Panics if no resource with the id exists.
     pub fn fetch_id_mut(&self, id: TypeId, comp_id: usize) -> FetchIdMut {
-        let c = self.fetch_internal(id, comp_id);
+        let c = self.try_fetch_internal(id, comp_id).expect(RESOURCE_NOT_FOUND);
```

Comment:
Sorry for requesting changes once more, but I think you forgot to change this one to `try_fetch_id_mut().expect(...)`. And yes, please squash your commits afterwards.